### PR TITLE
Issue #3171841 by chmez: As a CM+ I want to have the possibility to create a button in the "Hero" section to the sign-up page

### PR DESCRIPTION
**Pull request by ghost**
## Problem
When CM tries to use the sign-up page as the target of a button from the "Hero" section then CM receiving the following error message: "The path '/user/register' is inaccessible.".

## Solution
Give the "Link to any page" permission for CM+.

## Issue tracker
- https://www.drupal.org/project/social/issues/3171841
- https://getopensocial.atlassian.net/browse/YANG-3704

## How to test
- [ ] Enable the "Social Landing Page" module.
- [ ] Log in as CM and navigate to the "/node/add/landing_page" page.
- [ ] Press "Add Section" and press "Add Hero" and fill "/user/register" in the "URL" field of the "Link for Anonymous User" section.
- [ ] Fill other required fields and press "Save".
- [ ] You shouldn't see the following error message on the top of the page: "1 error has been found: Link for Anonymous User".
- [ ] You shouldn't see the following error message to each field in the "Link for Anonymous User" section: "The path '/user/register' is inaccessible.".

## Release notes
Allow using any URLs in link fields by CM+.

### CODE REVIEW

**Review by ribel — APPROVED** [review]
Looks good to me

**Comment by ribel** [comment]
🍒 picked to 8.x and 9.x

File: modules/social_features/social_landing_page/social_landing_page.install [hunk 1]
```diff
@@ -657,3 +658,12 @@ function social_landing_page_update_8808() {
     \Drupal::service('module_installer')->install(['social_book_featured']);
   }
 }
+
+/**
+ * Allow using any URLs in link fields by CM+.
+ */
+function social_landing_page_update_8809() {
+  foreach (['contentmanager', 'sitemanager'] as $role) {
+    user_role_grant_permissions($role, ['link to any page']);
```
Comment (ronaldtebrake) [inline modules/social_features/social_landing_page/social_landing_page.install:667]: Hi @OH17L 
After checking this in our latest update I do notice some implications especially for our Tier 0+ customers.
This permission allows CM/SM to see menu links which they don't have access to. 
The menu link also gives access denied once visited.

Screenshot 2020-10-06 at 15 51 41

In this screenshot you see what it looks like for SM when this permission is disabled:
Screenshot 2020-10-06 at 15 48 34

Comment (OH17L) [reply modules/social_features/social_landing_page/social_landing_page.install:667]: I created a task for this [here](https://getopensocial.atlassian.net/browse/YANG-3924).

**Pull request merged**